examples/plot_incr_kmeans.py

```diff
@@ -0,0 +1,319 @@
+"""
+Incremental KMeans
+==================
+
+In an active learning setting, the trade-off between exploration and
+exploitation plays a central role. Exploration, or diversity, is
+usually enforced using coresets or, more simply, a clustering algorithm.
+KMeans is therefore used to select samples that are spread across the
+dataset in each batch. However, to the best of our knowledge, maintaining diversity
+throughout the whole experiment is something that has not been considered.
+
+By allowing to start the optimization with fixed cluster centers, our
+Incremental KMeans allows for an optimal exploration of the space since
+samples are less likely to be selected nearby an already selected sample.
+
+This notebook introduces the principle behind Incremental KMeans and runs
+a small benchmark on generated data.
+
+"""
+
+from sklearn.cluster import MiniBatchKMeans
+from sklearn.metrics import pairwise_distances_argmin_min
+from sklearn.ensemble import RandomForestClassifier
+from sklearn.model_selection import train_test_split
+from cardinal.kmeans import IncrementalMiniBatchKMeans
+from cardinal.clustering import MiniBatchKMeansSampler, IncrementalMiniBatchKMeansSampler
+from sklearn.datasets import make_blobs
+from matplotlib import pyplot as plt
+from scipy.stats import ttest_rel
+from cardinal.utils import ActiveLearningSplitter
+import numpy as np
+
+# Inertia
+# ^^^^^^^
+#
+# We will be talking a  lot about inertia in this notebook which is the
+# euclidean distance from the points in a dataset to their closest center.
+# We define a function to measure it.
+
+
+def inertia(data, centers):
+    return (pairwise_distances_argmin_min(data, centers)[1] ** 2).sum()
+
+
+
+# Active learning like-experiment
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+#
+# One core element of the experiment differs from active learning. In fact,
+# in active learning, we do not expect to select a number of samples equal to
+# the number of real cluster in the data. We also build our clustering over
+# several iterations. We now try to do the same on generated data.
```

Comment:
Not sure what this means as what's below looks like an AL experiments. Is this a residue from the previous organisation of the example ?

Reply:
Exactly